### PR TITLE
Adds extra spess ruins

### PR DESCRIPTION
**Pull request by Nazsgull**
## What Does This PR Do
Adds the following space ruins:

> Syndicakes Factory
> debris1
> debris2
> debris3

## Why It's Good For The Game

Adds more variety to the pool of possible space ruins you can find when spacefaring. This can lead to new roleplay options, like discovering the terrible secret of the Syndicakes or finding a very useful pill bottle lying around.
## Images of changes
Please notify me If any of the following cannot be visualized.

**Syndicakes Factory**:
_"Syndicate used to get funds selling corgi cakes produced here. Was it hit by meteors or by a Nanotrasen comando?"_
This ruin is built like a puzzle, there's a room with loot and machines that will only work if powered. To do that you have to gain access to the northern room, fight a very angry corgi (alien sentinel with corgi skin) and feed the generator with the plasma sheets. Once powered, the room has an autolathe, a telecomms array, and suit storage containing a syndicate suit.
Syndicakesv1

**debris1**:
"A bunch of metal chunks, wires and space waste"
debris1v1
Loot:
- Random toolbox
- Random tool x2
- Metal
- Rods

**debris2**:
"A bunch of metal chunks, wires and space waste that used to be some kind of secure storage facility"
debris2v1
Loot (space):
- Biosuit helmet
- Biosuit
- Large beaker
- Metal
- Rods
Inside (vault zone):
- Bluespace beaker
- Bananium sheets x5
- Refined bluespace crystal x3
- DNA injector (clownchat)

**debris3**:
debris3v1
Loot:
- Variety pills bottle ()
- Arcade machine board
- Random tool x2
- Random official poster
- Random contraband poster
- Glass shards
- Metal
- Rods

**Mazes have been removed, might come back in the form of a Gateway Mission, soon™.**

## Changelog
:cl:
add: Adds more ruins you can encounter while spacefaring:
add: Syndicakes Factory
add: debris1
add: debris2
add: debris3
/:cl:

### CODE REVIEW

**Comment by ZomgPonies** [comment]
Mazes would work better as gateway destinations due to the fact they take a whole z-level and are only accessible from one side.

**Comment by SabreML** [comment]
To fix the map compilation issue, run `Prepare Maps.bat` with the changes already in, then `mapmerge.bat` on the added ruins. The Mapmerge tool needs a 'before' file to reference, so this just tricks it into working.

**Comment by taukausanake** [comment]
Wow those mazes. Yeah, they would definitely go better as away missions but the problem with that is they can be traced, ruins or missions, and cheesed easily. I barely see any difference between the images provided but that could work slightly with confusing the players with them. They also look pretty cubical, at least half of the map, so you could probably work it out in sections. Not well designed in my opinion

**Comment by Kyep** [comment]
Factory/debris1/debris2 are fine.
Debris3: what's in the variety pillbottle?
Mazes: might be better as a big gateway mission. Also, how do you stop people just welding through the walls?

**Comment by AffectedArc07** [comment]
Im really not sold on the whole maze ones. If they are pre-generated each time, someone will just draw the line in the map editor and figure it out. It also feels really out of place in space since its just a square, not an asteroid or anything. Other two seem like good concepts though.

**Comment by Kyep** [comment]
AA is right. Static mazes aren't really viable. People will memorize the solutions, or have the maps up on a second monitor. Probably best to axe the mazes completely from this PR.

**Comment by Nazsgull** [comment]
Axing the mazes, probably will add another thing to replace them, something not that massive

**Comment by Nazsgull** [comment]
> To fix the map compilation issue, run `Prepare Maps.bat` with the changes already in, then `mapmerge.bat` on the added ruins. The Mapmerge tool needs a 'before' file to reference, so this just tricks it into working.
> 
> Edit: Also, you seem to have a loose map file, `enygma.dmm` is in the main directory.

What do you mean with "before file"?

**Comment by SabreML** [comment]
> What do you mean with "before file"?

`Prepare Maps.bat` makes a backup of all the map files, and `mapmerge.bat` looks for changes between the 'before' and 'after' versions, and tidies up the code around them.
That way, if you're just editing one item, it only changes the part of the file with said item in it.
Saving the map through DM remakes the entire file, which makes Travis unhappy.

Since you're adding new maps, there's no 'before' file to reference from.
So,
 1. Run `Prepare Maps.bat` on the branch with the new ruins, this'll make a copy of the files to reference from later.

 2. Run `mapmerge.bat` and select the new files.
It'll look for changes between the 'before' and 'after' states, find none, and tidy the code up for you. Which sould fix the travis issue.

**Comment by AffectedArc07** [comment]
```
_maps/map_files/RandomRuins/SpaceRuins/debris1.dmm
_maps/map_files/RandomRuins/SpaceRuins/debris2.dmm
_maps/map_files/RandomRuins/SpaceRuins/debris3.dmm
_maps/map_files/RandomRuins/SpaceRuins/mazeA.dmm
_maps/map_files/RandomRuins/SpaceRuins/mazeB.dmm
_maps/map_files/RandomRuins/SpaceRuins/mazeC.dmm
```
This can be fixed with `tools/mapmerge2/dmm2tgm.bat`

```
loading ./map_files/RandomRuins/SpaceRuins/syndiecakesfactory.dmm
_maps/map_files/RandomRuins/SpaceRuins/syndiecakesfactory.dmm:389:error: bad type or variable: corgiGibber
_maps/map_files/RandomRuins/SpaceRuins/syndiecakesfactory.dmm:389:error: expected )
Failed to load map file /home/travis/build/ParadiseSS13/Paradise/_maps/map_files/RandomRuins/SpaceRuins/syndiecakesfactory.dmm.
```
This is likely an issue with var settings

**Review by SabreML — CHANGES_REQUESTED** [review]
The SMES in the Syndicakes Factory doesn't get power from the generator in the current configuration:
Capture1

The terminal is the power input, so it'd work better like this:
Capture

Wires:
Capture2

File: code/datums/ruins/space.dm [hunk 1]
```diff
@@ -248,3 +248,58 @@
...
 	always_place = TRUE // Its designed to make exploring other space ruins more accessible
 	cost = 0 // Force spawned so shouldnt have a cost
+
+/datum/map_template/ruin/space/syndiecakesfactory
+	id = "Syndiecakes Factory"
+	suffix = "syndiecakesfactory.dmm"
+	name = "Syndicakes Factory"
+	description = "Syndicate used to get funds selling corgi cakes produced here. Was it hit by meteors or by a Nanotrasen comando?"
+	allow_duplicates = FALSE //just in case
+	always_place = FALSE
```
Comment (AffectedArc07) [inline code/datums/ruins/space.dm:258]: Default to false

File: code/datums/ruins/space.dm [hunk 2]
```diff
@@ -248,3 +248,58 @@
...
+	cost = 2 //telecomms + multiple mobs
+
+/datum/map_template/ruin/space/debris1
+	id = "debris1"
+	suffix = "debris1.dmm"
+	name = "Debris field 1"
+	description = "A bunch of metal chunks, wires and space waste"
+	allow_duplicates = TRUE
```
Comment (AffectedArc07) [inline code/datums/ruins/space.dm:266]: Defaults to true

File: _maps/map_files/RandomRuins/SpaceRuins/syndiecakesfactory.dmm [hunk 3]
```diff
@@ -0,0 +1,2291 @@
...
+/area/ruin/unpowered)
+"Uq" = (
+/obj/machinery/conveyor/east{
+	id = null
+	},
+/turf/simulated/floor/plasteel,
+/area/ruin)
+"UC" = (
+/obj/machinery/tcomms/relay/ruskie{
+	network_id = "STORAGE-RELAY"
```
Comment (AffectedArc07) [inline _maps/map_files/RandomRuins/SpaceRuins/syndiecakesfactory.dmm:534]: Change this ID to something else.

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]
Few things

**Comment by AffectedArc07** [comment]
```
$ tools/travis/check_grep.sh
_maps/map_files/RandomRuins/SpaceRuins/mazeA.dmm
_maps/map_files/RandomRuins/SpaceRuins/mazeB.dmm
_maps/map_files/RandomRuins/SpaceRuins/mazeC.dmm
ERROR: Non-TGM formatted map detected. Please convert it using Map Merger!
The command "tools/travis/check_grep.sh" exited with 1.
```
Still fails on maze

File: code/datums/ruins/space.dm (hunk 1 above)
Comment (Nazsgull) [reply code/datums/ruins/space.dm:258]: Which one?

Comment (AffectedArc07) [reply code/datums/ruins/space.dm:258]: `always_place`

File: _maps/map_files/RandomRuins/SpaceRuins/syndiecakesfactory.dmm (hunk 3 above)
Comment (Nazsgull) [reply _maps/map_files/RandomRuins/SpaceRuins/syndiecakesfactory.dmm]: Done

File: code/datums/ruins/space.dm (hunk 2 above)
Comment (Nazsgull) [reply code/datums/ruins/space.dm:266]: Same as above, what do you mean?

**Comment by AffectedArc07** [comment]
```
loading ./map_files/RandomRuins/SpaceRuins/mazeC.dmm
_maps/map_files/RandomRuins/SpaceRuins/mazeC.dmm:8:error: bad type or variable: To
_maps/map_files/RandomRuins/SpaceRuins/mazeC.dmm:8:error: expected )
```

**Comment by Nazsgull** [comment]
Allright, Travis is now less angry (linters do pass yay!) but there's a "To" var messing up mazeC somewhere, I'll look into it tomorrow.

File: code/datums/ruins/space.dm (hunk 1 above)
Comment (Nazsgull) [reply code/datums/ruins/space.dm]: solved

File: code/datums/ruins/space.dm (hunk 2 above)
Comment (Nazsgull) [reply code/datums/ruins/space.dm:265]: solved

**Comment by AffectedArc07** [comment]
CL is malformed, please edit
image
:cl:
add: New ruins (a, b, c, etc)
/ :cl:

File: code/datums/ruins/space.dm [hunk 4]
```diff
@@ -248,3 +248,35 @@
...
+	description = "Syndicate used to get funds selling corgi cakes produced here. Was it hit by meteors or by a Nanotrasen comando?"
+
+	cost = 2 //telecomms + multiple mobs
+
+/datum/map_template/ruin/space/debris1
+	id = "debris1"
+	suffix = "debris1.dmm"
+	name = "Debris field 1"
+	description = "A bunch of metal chunks, wires and space waste"
+	allow_duplicates = TRUE
```
Comment (AffectedArc07) [inline code/datums/ruins/space.dm:265]: `allow_duplicates` defaults to `TRUE`

File: code/datums/ruins/space.dm [hunk 5]
```diff
@@ -248,3 +248,35 @@
...
+	cost = 1
```
Comment (AffectedArc07) [inline code/datums/ruins/space.dm:266]: `cost` defaults to `1`

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]
Please remove the overrides for `allow_duplicates` and `cost`

File: code/datums/ruins/space.dm (hunk 4 above)
Comment (Nazsgull) [reply code/datums/ruins/space.dm]: Removed

File: code/datums/ruins/space.dm (hunk 5 above)
Comment (Nazsgull) [reply code/datums/ruins/space.dm]: Removed

**Review by AffectedArc07 — APPROVED** [review]
Code side is fine, though ill let a head have a final vote on the maps

**Comment by Nazsgull** [comment]
> Code side is fine, though ill let a head have a final vote on the maps

Understood

**Review by Kyep — APPROVED** [review]

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**